front/src/modules/settings/profile/components/DeleteWorkspace.tsx

```diff
@@ -0,0 +1,118 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import { AnimatePresence, LayoutGroup } from 'framer-motion';
+import { useRecoilValue } from 'recoil';
+
+import { currentUserState } from '@/auth/states/currentUserState';
+import { Button } from '@/ui/button/components/Button';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { Modal } from '@/ui/modal/components/Modal';
+import { SubSectionTitle } from '@/ui/title/components/SubSectionTitle';
+
+import { debounce } from '../../../../utils/debounce';
+
+const StyledCenteredButton = styled(Button)`
+  justify-content: center;
+`;
+
+const StyledDeleteButton = styled(StyledCenteredButton)`
+  border-color: ${({ theme }) => theme.color.red20};
+  color: ${({ theme }) => theme.color.red};
+  font-size: ${({ theme }) => theme.font.size.md};
+  line-height: ${({ theme }) => theme.text.lineHeight.lg};
+`;
+
+const StyledTitle = styled.div`
+  font-size: ${({ theme }) => theme.font.size.lg};
+  font-weight: ${({ theme }) => theme.font.weight.semiBold};
+`;
+
+function EmailField({
+  email,
+  setEmail,
+  userEmail,
+  isDeleteDisabled,
+}: {
+  email: string;
+  setEmail: Dispatch<SetStateAction<string>>;
+  userEmail?: string;
+  isDeleteDisabled: boolean;
```

Comment:
The naming shouldn't contain references to delete because this button shouldn't be aware of its parent logic. Maybe build the errorMessage in the parent ? Maybe we could also use TextInput directly in DeleteWorkspace.

Reply:
This is a good idea @lucasbordeau 🙇